pkg/query-service/app/preferences/model.go

```diff
@@ -0,0 +1,598 @@
+				return fmt.Errorf("error in adding bootstrap preference: %s", err.Error())
+			}
+		} else if err != nil {
+			// if there is some other error in query return from here
+			return fmt.Errorf("error in bootstrap entries in preference entity: %s", err.Error())
+		}
+
+	}
+
+	// create the user preference table
+	table_schema = `
+	PRAGMA foreign_keys = ON;
+	CREATE TABLE IF NOT EXISTS user_preference(
+		preference_id TEXT NOT NULL,
+		preference_value TEXT,
+		user_id TEXT NOT NULL,
+		PRIMARY KEY (preference_id,user_id),
+		FOREIGN KEY (preference_id)
+			REFERENCES preference(id)
+			ON UPDATE CASCADE 
+			ON DELETE CASCADE,
+		FOREIGN KEY (user_id)
+			REFERENCES users(id)
+			ON UPDATE CASCADE
+			ON DELETE CASCADE
+	);`
+
+	_, err = db.Exec(table_schema)
+	if err != nil {
+		return fmt.Errorf("error in creating user_preference table: %s", err.Error())
+	}
+
+	// create the org preference table
+	table_schema = `
+	PRAGMA foreign_keys = ON;
+	CREATE TABLE IF NOT EXISTS org_preference(
+		preference_id TEXT NOT NULL,
+		preference_value TEXT,
+		org_id TEXT NOT NULL,
+		PRIMARY KEY (preference_id,org_id),
+		FOREIGN KEY (preference_id)
+			REFERENCES preference(id)
+			ON UPDATE CASCADE 
+			ON DELETE CASCADE,
+		FOREIGN KEY (org_id)
+			REFERENCES organizations(id)
+			ON UPDATE CASCADE
+			ON DELETE CASCADE
+	);`
+
+	_, err = db.Exec(table_schema)
+	if err != nil {
+		return fmt.Errorf("error in creating org_preference table: %s", err.Error())
+	}
+
+	// if there is no error then return nil
+	return nil
+}
+
+func GetUserPreference(ctx context.Context, preferenceId string, orgId string) (*PreferenceKV, *model.ApiError) {
+	userPreference := PreferenceKV{}
+	orgPreference := PreferenceKV{}
+	user := common.GetUserFromContext(ctx)
+
+	// get the preference id and value from the user preference table
+	query := `SELECT preference_id, preference_value FROM user_preference WHERE preference_id = $1 AND user_id = $2;`
+	err := db.Get(&userPreference, query, preferenceId, user.Id)
+
+	if err != nil && err != sql.ErrNoRows {
+		return nil, &model.ApiError{Typ: model.ErrorExec, Err: err}
+	}
+
+	// get the preference id and value from the org preference table
+	query = `SELECT preference_id,preference_value FROM org_preference WHERE preference_id=$1 AND org_id=$2;`
+	err = db.Get(&orgPreference, query, preferenceId, orgId)
+
+	if err != nil && err != sql.ErrNoRows {
+		return nil, &model.ApiError{Typ: model.ErrorExec, Err: err}
+	}
+
+	// get the details for the preference entity
+	preference := Preference{}
+	query = `SELECT user, org , default_value FROM preference WHERE id = $1;`
+	err = db.Get(&preference, query, preferenceId)
+
+	// return if unable to fetch the preference entity as we won't be sure about preference being enabled or not also if the preference doesn't exist
+	if err != nil {
+		return nil, &model.ApiError{Typ: model.ErrorNotFound, Err: fmt.Errorf("error while fetching the preference entity: %s", preferenceId)}
+	}
```

Comment:
Return separate errors for the non-existent preference and for unknown errors.

Reply:
done